packages/nodes-base/nodes/Google/BigQuery/v2/actions/database/executeQuery.operation.ts

```diff
@@ -401,7 +402,10 @@ export async function execute(this: IExecuteFunctions): Promise<INodeExecutionDa
 
 	let waitTime = 1000;
 	while (jobs.length > 0) {
-		const completedJobs: string[] = [];
+		if (abortSignal?.aborted) {
+			break;
+		}
+		const settledJobs: string[] = [];
```

Comment:
I will still suggest to do this in `for` loop to catch `aborted` during iteration
you could name outer `while` loop and break it in `for` loop
```
outerLoop: while (

...
in for
break outerLoop;
```